prov/util/src/util_shm.c

```diff
@@ -208,11 +216,28 @@ int smr_map_create(const struct fi_provider *prov, int peer_count,
 	return 0;
 }
 
+static int smr_match_name(struct dlist_entry *item, const void *args)
+{
+	return !strcmp(container_of(item, struct smr_ep_name, entry)->name,
+		       (char *) args);
+}
+
 int smr_map_to_region(const struct fi_provider *prov, struct smr_peer *peer_buf)
 {
 	struct smr_region *peer;
 	size_t size;
 	int fd, ret = 0;
+	struct dlist_entry *entry;
+
+	pthread_mutex_lock(&ep_list_lock);
+	entry = dlist_find_first_match(&ep_name_list, smr_match_name,
+				       peer_buf->peer.name);
+	pthread_mutex_unlock(&ep_list_lock);
+	if (entry) {
```

Comment:
As long as you're adding locks around the ep_name_list, I would continue to hold the lock while entry is being accessed.  (There's no reference being held on entry.)

Reply:
Updated